### PR TITLE
WT-13317 Moving bash script to separate files - code coverage

### PR DESCRIPTION
**Pull request by bzm-jas**

### CODE REVIEW

File: test/coverage-report
Comment (jiechenbo) [inline test/coverage-report:69]: Style: Add a line to the end of the script

File: test/coverage-report
Comment (jiechenbo) [inline test/coverage-report:2]: Style: Add a .sh postfix to the name of all files file to denote it is a bash scripts

File: test/coverage-report
Comment (jiechenbo) [inline test/coverage-report:7]: Can we add a check that the script requires four arguments? If four arguments are not supplied we fail it

**Review by jiechenbo — COMMENTED** [review]
@bzm-jas thanks for doing the work! Very close just a few comments 👍

**Comment by jiechenbo** [comment]
@bzm-jas could you also refactor the "code coverage analysis" function to a script too?

**Review by jeremythorp — CHANGES_REQUESTED** [review]
Thanks Jasmine. I've added some comments, and I think this change is most of the way there.

File: test/coverage-report.sh
Comment (bzm-jas) [reply test/coverage-report.sh:4]: The fourth argument is "optional" since we wouldn't have pr args for patch builds and other local changes, so it can be an empty string that is not detected by the script. I've added another check to make sure that we are either getting 3 or 4 arguments to make sure the script fails straight away if the incorrect number of arguments are given. see [f6c41f6](https://github.com/wiredtiger/wiredtiger/pull/10915/commits/f6c41f665c9227651419aa3517491c20f8e1f7f3)

File: test/coverage-report.sh
Comment (bzm-jas) [reply test/coverage-report.sh:21]: I've added a check to ensure the correct number of arguments are being parsed in and added double quotes for pr_args to parse it as one string and making sure the script captures it correctly as one argument, see: [f6c41f6](https://github.com/wiredtiger/wiredtiger/pull/10915/commits/f6c41f665c9227651419aa3517491c20f8e1f7f3)

**Comment by bzm-jas** [comment]
> 892ef4a

refactored code coverage analysis in [892ef4a](https://github.com/wiredtiger/wiredtiger/pull/10915/commits/892ef4aed53fd969412474794704346d3208ebd4)

File: test/evergreen/coverage-report.sh
Comment (jiechenbo) [inline test/evergreen/coverage-report.sh:7]: I would just combine both if conditions in this case.

File: test/evergreen/code_coverage_analysis.sh
Comment (jiechenbo) [inline test/evergreen/code_coverage_analysis.sh:14]: I would become more explicit and strict in the sense of 
```
if [  combine_coverage_report == True && $# != 7 ]
```

Is it possible to also move the printing of the usage to a function such as print_usage(), so we can re-use it in both if conditions?

File: test/evergreen/coverage-report.sh
Comment (bzm-jas) [reply test/evergreen/coverage-report.sh]: if statements combined in [eb83923](https://github.com/wiredtiger/wiredtiger/pull/10915/commits/eb83923db61711b2bf4f900af80ff204588617ee)

File: test/evergreen/code_coverage_analysis.sh
Comment (bzm-jas) [reply test/evergreen/code_coverage_analysis.sh]: good idea, I fixed the if statement and added print_usage in the most recent commit: [eb83923](https://github.com/wiredtiger/wiredtiger/pull/10915/commits/eb83923db61711b2bf4f900af80ff204588617ee)

**Review by jiechenbo — APPROVED** [review]
LGTM! Make sure to run the code coverage tests at the end before merging

File: test/evergreen/code_coverage_analysis.sh
Comment (jeremythorp) [inline test/evergreen/code_coverage_analysis.sh:18]: Hi @bzm-jas , if I've understood correctly, `combine_coverage_report` is assigned a value at line 27, but is first referenced here at line 18. At this point the `combine_coverage_report` won't have a value, and so this test will always fail.

**Review by jeremythorp — CHANGES_REQUESTED** [review]
Hi @bzm-jas , I think it's nearly there, but one further change is required for how `combine_coverage_report` is initialised and used. Thanks, Jeremy

File: test/evergreen/code_coverage_analysis.sh
Comment (bzm-jas) [reply test/evergreen/code_coverage_analysis.sh]: Hi Jeremy, thanks for picking this up, I have fixed it in the most recent commit: [8a1b0de](https://github.com/wiredtiger/wiredtiger/pull/10915/commits/8a1b0def5956ccb3d886541a6ef7a8a5fafc3ad3)

Comment (jeremythorp) [reply test/evergreen/code_coverage_analysis.sh]: Thanks

**Review by jeremythorp — APPROVED** [review]
Just need to get `s_all` happy again, then it LGTM.

**Pull request merged**
